src/pyfme/aircrafts/aircraft.py

```diff
@@ -87,26 +99,38 @@ def update(self, controls, system, environment):
                 self.controls[control_name] = control_value
             else:
                 # TODO: maybe raise a warning and assign max deflection
-                msg = "Control {} out of range ({} when max={} and min={" \
-                      "}".format(control_name, limits[1], limits[0])
+                msg = ("Control {} out of range ({} when max={} and min={"
+                       "})".format(control_name, control_value, limits[1],
+                                   limits[0]))
                 raise ValueError(msg)
 
+    def _calculate_aerodynamics(self, system, environment):
+
         # Velocity relative to air: aerodynamic velocity.
         aero_vel = system.vel_body - environment.body_wind
 
         self.alpha, self.beta, self.TAS = calculate_alpha_beta_TAS(
-            u=aero_vel[0], v=aero_vel[1], w=aero_vel[2])
+            u=aero_vel[0], v=aero_vel[1], w=aero_vel[2]
+        )
+
+        # Setting velocities & dynamic pressure
+        self.CAS = tas2cas(self.TAS, environment.p, environment.rho)
+        self.EAS = tas2eas(self.TAS, environment.rho)
+        self.Mach = self.TAS / environment.a
+        self.q_inf = 0.5 * environment.rho * self.TAS ** 2
+
+    def _calculate_aerodynamics_2(self, TAS, alpha, beta, environment):
```

Comment:
Where is this function used?

Reply:
Aircraft class is to be refactored, will have a look at it when this is merged.